cpp/src/parquet/arrow/schema.cc

```diff
@@ -1017,7 +1017,9 @@ Result<bool> ApplyOriginalMetadata(const Field& origin_field, SchemaField* infer
 
       // Restore extension type, if the storage type is the same as inferred
       // from the Parquet type
-      if (ex_type.storage_type()->Equals(*inferred->field->type())) {
+      if (ex_type.storage_type()->Equals(*inferred->field->type()) ||
+          (ex_type.extension_name() == "arrow.json" &&
+           !ex_type.storage_type()->Equals(*inferred->field->type()))) {
```

Comment:
This "types are not equal" check is superfluous? Because if they were equal, the first part of the if check would have passed and we wouldn't get here?

I assume that in case of JSON extension type, we either just have to assume that the data will correctly use string, or we check explicitly that the inferred type is string type?

Reply:
It is superfluous indeed. Changed as per your suggestion: https://github.com/apache/arrow/pull/44215/commits/3364c0f0f70a37c84e134043537c4ee4d55fbbd6